### PR TITLE
Update ArrayJob for Map task changes

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
Backwards compatible change for array jobs now that `container_array` task types with `task_type_version > 0` do not predetermine array job size. Instead they will pass in the `min_success_ratio` which can be used at runtime to determine the `min_successes`

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/609

## Follow-up issue
_NA_

### CODE REVIEW

File: protos/flyteidl/plugins/array_job.proto [hunk 1]
```diff
@@ -14,11 +14,16 @@ message ArrayJob {
 
     // Defines the number of instances to launch at most. This number should match the size of the input if the job
     // requires processing of all input data. This has to be a positive number.
-    // +required.
     int64 size = 2;
```
Comment (EngHabu) [inline protos/flyteidl/plugins/array_job.proto:17]: nit: add a comment here explaining that if not defined, the backend will read the inputs to determine the size... or something..

**Review by EngHabu — APPROVED** [review]

File: protos/flyteidl/plugins/array_job.proto (hunk 1 above)
Comment (katrogan) [reply protos/flyteidl/plugins/array_job.proto:18]: sure, done @EngHabu

**Review by EngHabu — APPROVED** [review]
